ground/src/main/java/com/google/android/ground/ui/offlineareas/selector/OfflineAreaSelectorViewModel.kt

```diff
@@ -98,15 +100,21 @@ internal constructor(
     isDownloadProgressVisible.value = true
     downloadProgress.value = 0f
     viewModelScope.launch(ioDispatcher) {
-      offlineAreaRepository.downloadTiles(viewport!!).collect { (bytesDownloaded, totalBytes) ->
-        val progressValue =
-          if (totalBytes > 0) {
-            (bytesDownloaded.toFloat() / totalBytes.toFloat()).coerceIn(0f, 1f)
-          } else {
-            0f
-          }
-        downloadProgress.postValue(progressValue)
-      }
+      offlineAreaRepository
+        .downloadTiles(viewport!!)
+        .catch {
+          isFailure.postValue(true)
+          isDownloadProgressVisible.postValue(false)
+        }
+        .collect { (bytesDownloaded, totalBytes) ->
+          val progressValue =
```

Comment:
Could you please refactor the logic in this lambda into its own function while you're at it?